### PR TITLE
Large screen style

### LINKED ISSUE #1536: Add CSS styles for min-width beyond 1200px
### Requirements

- [X] This is a feature request and not a bug report. Otherwise, please create a new [bug report](https://github.com/LemmyNet/lemmy-ui/issues/new?assignees=&labels=bug%2Ctriage&projects=&template=BUG_REPORT.yml) instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this request (or a similar one) already exists.
- [X] It's a single feature. Please don't request multiple features in one issue.

### Describe the feature you'd like

I've got a browser window width spanning just over 2500 pixels.  The existing `@media (min-width: 1200px)` styles are rather limiting, considering how `.container` and its mates are forced to a `max-width` of `1140px`.  So, over half of my browser window on the front page is empty space.

Just getting rid of the `max-width` on the CSS solves part of it.  But, there's also other `.col-*` properties that could re-adjusted with their `flex` and `max-width` columns.  The right-pane needs to be shrunk, possibly even to a fix size of 300px or so, and the left-pane expanded as far as it can.

### PR DESCRIPTION
**Pull request by SleeplessOne1917**
Closes #1536. Here is what the pages look like now:

# Home
image

# Community Page
image

# Communities
image

# Create Post
image

# Create Community
image

# Search
image

# Admin Settings
image

# Inbox
image

# Reports
image

# Modlog
image

# Instances
image

I still have to do settings and profile. Will update the PR with screenshots when done.

### CODE REVIEW

**Comment by jsit** [comment]
Hm it seems like some of these changes a are unrelated to screen widening?

**Comment by SleeplessOne1917** [comment]
I also removed a bit of unnecessary markup.

File: src/shared/components/person/verify-email.tsx [hunk 1]
```diff
@@ -53,20 +53,18 @@ export class VerifyEmail extends Component<any, State> {
...
-              </h5>
-            )}
-          </div>
+        <div className="col-12 col-lg-6 offset-lg-3 mb-4">
+          <h5>{I18NextService.i18n.t("verify_email")}</h5>
+          {this.state.verifyRes.state === "loading" && (
+            <h5>
+              <Spinner large />
+            </h5>
+          )}
```
Comment (SleeplessOne1917) [inline src/shared/components/person/verify-email.tsx:67]: All this page displays is a title and a loading spinner. @dessalines should we still keep this file?

File: src/assets/css/main.css [hunk 2]
```diff
@@ -276,7 +277,10 @@ hr {
 }
 
 .mini-overlay {
-  display: block;
```
Comment (jsit) [inline src/assets/css/main.css:279]: This feels like it doesn't belong in this PR?

File: src/assets/css/main.css [hunk 3]
```diff
@@ -431,3 +435,15 @@ br.big {
...
+.comment,
+.markdown-textarea {
+  max-width: 50rem;
+}
+
+input[type="checkbox"],
+.form-check label,
+select,
+option {
+  cursor: pointer;
```
Comment (jsit) [inline src/assets/css/main.css:448]: This feels like it doesn't belong in this PR; plus I'm not sure pointers for these things is correct.

File: src/shared/components/comment/comment-nodes.tsx [hunk 4]
```diff
@@ -89,7 +89,11 @@ export class CommentNodes extends Component<CommentNodesProps, any> {
             "ms-1": !!this.props.isChild,
             "border-top border-light": !this.props.noBorder,
           })}
-          style={`border-left: 2px solid ${borderColor} !important;`}
+          style={
```
Comment (jsit) [inline src/shared/components/comment/comment-nodes.tsx:92]: This feels separate from this PR

File: src/shared/components/community/communities.tsx [hunk 5]
```diff
@@ -145,57 +145,41 @@ export class Communities extends Component<any, CommunitiesState> {
...
                           <CommunityLink community={cv.community} />
                         </td>
-                        <td className="text-right">
-                          {numToSI(cv.counts.subscribers)}
-                        </td>
-                        <td className="text-right">
-                          {numToSI(cv.counts.users_active_month)}
-                        </td>
-                        <td className="text-right d-none d-lg-table-cell">
+                        <td>{numToSI(cv.counts.subscribers)}</td>
```
Comment (jsit) [inline src/shared/components/community/communities.tsx:151]: Why remove `text-right`? This was intentional so numbers are lined up

**Review by jsit — CHANGES_REQUESTED** [review]

File: src/shared/components/community/communities.tsx (hunk 5 above)
Comment (SleeplessOne1917) [reply src/shared/components/community/communities.tsx:151]: I can make them right aligned again, but `text-right` isn't a valid bootstrap class in v5 regardless.

File: src/assets/css/main.css (hunk 3 above)
Comment (SleeplessOne1917) [reply src/assets/css/main.css:448]: Should selects not have the cursor be a pointer? The label? both?

File: src/assets/css/main.css (hunk 2 above)
Comment (SleeplessOne1917) [reply src/assets/css/main.css:279]: I can make it a separate PR if you think that's appropriate. Keeping it the old way makes the thumbnail icon show up incorrectly image thumbnails.

File: src/shared/components/comment/comment-nodes.tsx (hunk 4 above)
Comment (SleeplessOne1917) [reply src/shared/components/comment/comment-nodes.tsx:92]: I can make this a separate PR.

File: src/assets/css/main.css (hunk 3 above)
Comment (jsit) [reply src/assets/css/main.css:448]: I'm kind of a stickler in that I think the default cursor styles for form elements is handled correctly by browsers by default -- a standard `default` arrow cursor, like it is on every other form element in your operating system. The `pointer` style was designed for hyperlinks.

I know this convention has changed in recent years but I'm not sure it's necessarily the "right" thing to do.

File: src/shared/components/community/communities.tsx (hunk 5 above)
Comment (jsit) [reply src/shared/components/community/communities.tsx:151]: Oh ha you're right -- I had to do this in #1538 too

File: src/assets/css/main.css (hunk 2 above)
Comment (jsit) [reply src/assets/css/main.css:279]: Ah, I remember how I handled this now. I added these classes to the thing element wrapping it:

https://github.com/LemmyNet/lemmy-ui/pull/1570/files#diff-cb34074ed5916cfbee02ee325240476c1ee8dd515e9e4559ffff1eaeec039f53R50

I think this makes more sense, to absolutely position the wrapper element (link/button) and keep the icon itself just positioned naturally within it.

**Comment by jsit** [comment]
Post Listing images are now "portrait"-oriented, and are no longer top-aligned with the post title:
Screenshot 2023-06-25 at 1 11 43 PM

**Comment by jsit** [comment]
Kind of throwing a wrench into things here, but I'm not sure this is even desirable in the first place? I don't know that this makes good use of the extra space gained; it will mostly just make things further apart.

I could see changing the max-width to something else (in fact it's larger in Bootstrap 5, and I overwrote the value of `$max-width` when switching to BS5 to maintain the existing look), but indefinite width feels weird to me.

Screenshot 2023-06-25 at 1 14 21 PM

**Comment by jsit** [comment]
From the original issue:

> over half of my browser window on the front page is empty space.

There's still empty space; it's now just between the main and the sidebar.

I think this is this one user's personal preference and shouldn't necessarily be globally adopted.

**Comment by alectrocute** [comment]
> From the original issue:
> 
> > over half of my browser window on the front page is empty space.
> 
> There's still empty space; it's now just between the main and the sidebar.
> 
> I think this is this one user's personal preference and shouldn't necessarily be globally adopted.

I agree completely.

**Comment by SleeplessOne1917** [comment]
Any thoughts on how to make it better? I think there's still too much whitespace but also an improvement on what's there currently.

**Comment by jsit** [comment]
1. Increase the max-width of the whole site
2. Change the sidebar to be 3 columns, not 4, on wide

**Comment by SleeplessOne1917** [comment]
> Post Listing images are now "portrait"-oriented, and are no longer top-aligned with the post title:

I thought the top aligned listing images looked funny on post listings with more vertical space. I can change it back if you'd prefer.

**Comment by jsit** [comment]
Here's an example of another way we might address this: #1590

File: src/shared/components/search.tsx [hunk 6]
```diff
@@ -526,13 +526,8 @@ export class Search extends Component<any, SearchState> {
       creatorSearchOptions,
       searchCommunitiesLoading,
       searchCreatorLoading,
-      communitiesRes,
     } = this.state;
 
-    const hasCommunities =
```
Comment (jsit) [inline src/shared/components/search.tsx:532]: Why remove this?

**Comment by jsit** [comment]
It seems like there's a lot going on here that, while good improvements, don't really have to do with making things wider. Can these be moved to a separate chore PR?

File: src/shared/components/search.tsx (hunk 6 above)
Comment (SleeplessOne1917) [reply src/shared/components/search.tsx:532]: Doh. I'll put it back.

**Comment by SleeplessOne1917** [comment]
What specifically do you think doesn't belong in this PR? I've gotten rid of everything not related to displaying on larger screen sizes at this point.

**Comment by jsit** [comment]
Some of this looks a little wonky to me? This sidebar looks too small, and the comments aren't left-aligned with the post content

Screenshot 2023-06-25 at 3 58 30 PM

I'd kind of like to see #1590 merged in first, then some more fine-tuning done if necessary.

I also like how this PR makes some semantic fixes, but they make the diffs harder to read than they need to be for such a large PR.

It might be easier if this were broken up into smaller PRs for each page.
